packages/jaeger-ui/src/components/DependencyGraph/DAG.tsx

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2023 The Jaeger Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { ReactNode } from 'react';
+
+import { Digraph, LayoutManager, cacheAs } from '@jaegertracing/plexus';
+import { TEdge, TVertex } from '@jaegertracing/plexus/lib/types';
+import { TMeasureNodeUtils } from '@jaegertracing/plexus/src/Digraph/types';
+
+import './dag.css';
+
+type TProps = {
+  serviceCalls: {
+    parent: string;
+    child: string;
+    callCount: number;
+  }[];
+};
+
+type TState = {
+  nodes: TVertex[];
+  edges: TEdge[];
+};
+
+const renderNode = (vertex: TVertex<TVertex>): ReactNode => {
+  return (
```

Comment:
how is it possible for this function to never be called in the test?

Reply:
Yeah, this is what caused test coverage to decrease